nix/ext/pg_cron.nix

```diff
@@ -1,31 +1,121 @@
 { lib, stdenv, fetchFromGitHub, postgresql }:
 
-stdenv.mkDerivation rec {
-  pname = "pg_cron";
-  version = "1.6.4";
+let
+  allVersions = {
+    "1.3.1" = {
+      rev = "v1.3.1";
+      hash = "sha256-rXotNOtQNmA55ErNxGoNSKZ0pP1uxEVlDGITFHuqGG4=";
+      postPatch = ''
+        # Add necessary includes
+        substituteInPlace src/pg_cron.c \
+          --replace '#include "postgres.h"' '#include "postgres.h"
+      #include "commands/async.h"
+      #include "miscadmin.h"'
 
-  buildInputs = [ postgresql ];
+        # Update function calls to use PostgreSQL 15 APIs
+        substituteInPlace src/pg_cron.c \
+          --replace 'ProcessCompletedNotifies();' '/* ProcessCompletedNotifies removed */' \
+          --replace 'pg_analyze_and_rewrite(parsetree, sql, NULL, 0,NULL);' 'pg_analyze_and_rewrite_fixedparams(parsetree, sql, NULL, 0, NULL);'
+      '';
```

Comment:
@samrose Could we have the patches as a `.patch` file? We already do this for plv8 ([ref](https://github.com/supabase/postgres/blob/develop/nix/ext/0001-build-Allow-using-V8-from-system.patch)).

Changes are easier to see that way and applying the patch with the Nix `patches` attribute is a standard practice.

Reply:
resolved